docs/src/rules/no-shadow.md

```diff
@@ -168,6 +176,44 @@ function b() {}
 
 Because `const a` and `const b` in the `if` statement are before the declarations in the outer scope, they are correct.
 
+#### hoist: types
+
+Examples of **incorrect** code for the `{ "hoist": "types" }` option:
+
+::: incorrect
+
+```ts
+/*eslint no-shadow: ["error", { "hoist": "types" }]*/
+
+type Bar<Foo> = 1;
+type Foo = 1;
+```
+
+:::
+
+#### hoist: functions-and-types
+
+Examples of **incorrect** code for the `{ "hoist": "functions-and-types" }` option:
+
+::: incorrect
+
+```ts
+/*eslint no-shadow: ["error", { "hoist": "functions-and-types" }]*/
+
+// types
+type Bar<Foo> = 1;
+type Foo = 1;
+
+// functions
+if (true) {
+  const b = 6;
+}
+
+function b() {}
```

Comment:
![Screenshot 2025-05-11 132607](https://github.com/user-attachments/assets/c2f3b960-d6c1-4aac-bf0f-cace91131028)

Function declarations are not reported here and seems it is intended.

https://github.com/eslint/eslint/blob/f60f2764971a33e252be13e560dccf21f554dbf1/lib/rules/no-shadow.js#L291-L294